### PR TITLE
fix: Default address form title option not translatable

### LINKED ISSUE #11413: Default Address Form Title option not translatable
**Describe the bug**
Within the Address form component used for e.g. the shipping address, the title form field automatically adds a default option. This default option's display value is currently hardcoded in the component and cannot be translated.

**Tell us the version of Spartacus**
- Library version: 3.1.0

**To Reproduce**
1. Go to the spartacus demo store
2. Login / register
3. Go to page https://spartacus-demo.eastus.cloudapp.azure.com/electronics-spa/de/USD/my-account/address-book
4. See title dropdown having all entries, but the first one, translated to german

**Expected behavior**
The title dropdown either uses a translatable default option or offers an option to not show that default option at all

**Screenshots**
image

**Desktop (please complete the following information):**
 - OS: all
 - Browser all
 - Version all

**Smartphone (please complete the following information):**
 - Device: all
 - OS: all
 - Browser all
 - Version all

**Additional context**
The hardcoded default option gets added in the file `address-form.component.ts`:
image

### PR DESCRIPTION
**Pull request by bit172**
closes #11413

### CODE REVIEW

File: projects/assets/src/translations/en/address.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 export const address = {
   addressForm: {
     title: 'Title',
+    noneTitle: 'Title',
```
Comment (dydome) [inline projects/assets/src/translations/en/address.ts]:
```suggestion
    defaultTitle: 'Title',
```

File: projects/storefrontlib/src/cms-components/checkout/components/shipping-address/address-form/address-form.component.ts [hunk 2]
```diff
@@ -172,6 +168,19 @@ export class AddressFormComponent implements OnInit, OnDestroy {
     this.addresses$ = this.userAddressService.getAddresses();
   }
 
+  getTitles(): Observable<Title[]> {
+    return combineLatest([
+      this.translation.translate('addressForm.noneTitle'),
+      this.userService.getTitles(),
```
Comment (dydome) [inline projects/storefrontlib/src/cms-components/checkout/components/shipping-address/address-form/address-form.component.ts]: 'getTitles' is deprecated you have to use new lib `@spartacus/user` and new method from facade of user profile

**Review by dydome — CHANGES_REQUESTED** [review]
Please pull the latest develop changes with the new user library and update deprecated methods.

File: projects/storefrontlib/src/cms-components/checkout/components/shipping-address/address-form/address-form.component.ts (hunk 2 above)
Comment (bgambocjaviniar) [reply projects/storefrontlib/src/cms-components/checkout/components/shipping-address/address-form/address-form.component.ts]: He will revert back to his old one and use the deprecated method.

We shouldn't create such dependencies from our main lib with feature libs.

Also, the one who will move this to the feature-lib in the near future will update the dependencies accordingly.

File: docs/migration/4_0.md [hunk 3]
```diff
@@ -438,6 +438,8 @@ Lib: @spartacus/core
 Change: constructor parameter `checkoutDeliveryService: CheckoutDeliveryService` is removed.
 AddressFormComponent now uses the new address verification function from  `UserAddressService` called `verifyAddress` instead of the `verifyAddress` function from `CheckoutDeliveryService`.  `UserAddressService.verifyAddress` does not use the ngrx store under the hood.
 
+Change: constructor parameter `translation: TranslationService` is added
```
Comment (dydome) [inline docs/migration/4_0.md]: I will suggest using something more common from migration doc like
`- `TranslationService` is a new, required constructor dependency.`

and than additional description

**Review by dydome — CHANGES_REQUESTED** [review]

**Review by dydome — APPROVED** [review]

**Review by Platonn — APPROVED** [review]

**Pull request merged**